extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts

```diff
@@ -5,12 +5,17 @@
 
 import { Transform } from "@itwin/core-geometry";
 import { ArcGisSymbologyRenderer } from "./ArcGisSymbologyRenderer";
-import { ArcGisGeometryBaseRenderer } from "@itwin/core-frontend";
+import { ArcGisAttributeDrivenSymbology, ArcGisGeometryBaseRenderer } from "@itwin/core-frontend";
 
+/** @internal */
 export class ArcGisCanvasRenderer extends ArcGisGeometryBaseRenderer {
   private _context: CanvasRenderingContext2D;
   private _symbol: ArcGisSymbologyRenderer;
 
+  public override get attributeSymbology() {
+    return this._symbol.attributeDriven ? (this._symbol as unknown) as ArcGisAttributeDrivenSymbology : undefined;
```

Comment:
Motivation for this cast?

Reply:
If the `attributeDriven` returns true, I know I have a [ArcGisUniqueValueSymbologyRenderer](https://github.com/iTwin/itwinjs-core/blob/ea715eb692b2d8e2bdb330ad7685956230a19d55/extensions/map-layers-formats/src/ArcGisFeature/ArcGisSymbologyRenderer.ts#L136), but ALSO an `ArcGisAttributeDrivenSymbology`, which **matters the most to me** :
`export class ArcGisUniqueValueSymbologyRenderer extends ArcGisSimpleSymbologyRenderer implements ArcGisAttributeDrivenSymbology`

Typescript would let me cast it as a `ArcGisSimpleSymbologyRenderer` (the base abstract class), but not the `ArcGisAttributeDrivenSymbology` it implements.  So I have to make a 'reinterpret' cast.

I feel like typescript is limiting polymorphism here, but maybe I just don't understand how it's meant to be used?

Reply:
Did you try something like:
```
export abstract class ArcGisSymbologyRenderer {
  public abstract isAttributeDriven(): this is ArcGisAttributeDrivenSymbology;
 // ...
}
```
Then:
```
export class ArcGisCanvasRenderer extends ArcGisGeometryBaseRenderer {
  public override get attributeSymbology(): ArcGisAttributeDrivenSymbology | undefined {
    return this._symbol.isAttributeDriven() ? this._symbol : undefined;
  }
  // ...
}
```

Reply:
done, tx for the suggestion!